src/SourceLocator/SourceStubber/PhpStormStubs/CachingVisitor.php

```diff
@@ -86,18 +87,27 @@ public function enterNode(Node $node): ?int
         }
 
         if ($node instanceof Node\Expr\FuncCall) {
-            try {
-                ConstantNodeChecker::assertValidDefineFunctionCall($node);
-            } catch (InvalidConstantNode) {
-                return null;
-            }
-
             $argumentNameNode = $node->args[0];
             assert($argumentNameNode instanceof Node\Arg);
             $nameNode = $argumentNameNode->value;
             assert($nameNode instanceof Node\Scalar\String_);
             $constantName = $nameNode->value;
 
+            // Modify the value to constant('STDIN') call
```

Comment:
This comment needs to be expanded a bit: it is clear to me that we're replacing `STDIN` with `constant('STDIN')`, so that we force BetterReflection to resolve that constant fetch, but that requires explanation for posterity too

Reply:
I've tried to improve it.